Assets/Src/AuthController/CredentialProviders/FirebaseCredentialProvider.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Threading.Tasks;
+using Firebase.Auth;
+using Src.AuthController.CredentialProviders;
+using Src.AuthController.CredentialProviders.Google;
+using Src.AuthController.Exceptions;
+
+namespace Src.AuthController
+{
+    public class FirebaseCredentialProvider : IFirebaseCredentialProvider
+    {
+        public FirebaseCredentialProvider()
+        {
+        }
+        
+        public async Task<Credential> GetCredentialAsync(FirebaseAuthProviderType authProviderType)
+        {
+            switch (authProviderType)
+            {
+                case FirebaseAuthProviderType.Google:
+                    var googleCredentialProvider = new GoogleCredentialProvider();
```

Comment:
Consider moving logic related to creation of credential providers into separate class.